File: java/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer, Integer> freqMap = new HashMap<>();
        for (int num : nums) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }
        int maxLength = 0;
        for (int key : freqMap.keySet()) {
            if (freqMap.containsKey(key + 1)) {
                int totalLength = freqMap.get(key) + freqMap.get(key + 1);
                maxLength = Math.max(maxLength, totalLength);
            }
        }
        return maxLength;
    }
}

File: cpp/0594-longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> freq;
    for (int num : nums) {
        freq[num]++;
    }
    int maxLen = 0;
    for (auto& [key, value] : freq) {
        if (freq.count(key + 1)) {
            maxLen = max(maxLen, value + freq[key + 1]);
        }
    }
    return maxLen;
    }
};